ddtrace/contrib/flask/patch.py

```diff
@@ -279,6 +282,42 @@ def traced_start_response(status_code, headers):
     return traced_start_response
 
 
+@with_instance_pin
+def traced_stream_with_context(pin, wrapped, instance, args, kwargs):
+    """
+    Wrapper for flask.stream_with_context
+
+    This function wraps streamed responses.
+    Note - Responses are stream after the flask wsgi app is called. Therefore the span produced by this
+    wrapper will be a sibling span of flask.request.
+    """
+    # stream_with_context(...) copies the current request context before returning an iterable.
+    # To avoid raising a RuntimeError in the wsgi server we must wrap the iterable returned by
+    # this function: https://github.com/pallets/flask/blob/2.1.2/src/flask/helpers.py#L113
+    iterable = wrapped(*args, **kwargs)
+
+    resource = u" ".join((flask.request.method, flask.request.path))
+
+    def _wraped_generator():
+        # TO FIX: GeneratorExit is called when the generator returned stream_with_context(...) is garbage collected.
```

Comment:
We should be getting a `StopIteration` exception raised from `iterable.__next__()` when there is no more to iterate over, right?

Reply:
with the `with span: while True`, do we end up with always marking this span as an error?